numpy/lib/npyio.py

```diff
@@ -624,29 +634,15 @@ def _savez(file, args, kwds, compress, allow_pickle=True, pickle_kwargs=None):
     else:
         compression = zipfile.ZIP_STORED
 
-    zipf = zipfile_factory(file, mode="w", compression=compression)
-
-    # Stage arrays in a temporary file on disk, before writing to zip.
-    fd, tmpfile = tempfile.mkstemp(suffix='-numpy.npy')
-    os.close(fd)
-    try:
+    with zipfile_factory(file, mode="w", compression=compression) as zipf:
```

Comment:
I don't think your context handler is exception safe, so either needs a fix or this should still be in a try..except clause.